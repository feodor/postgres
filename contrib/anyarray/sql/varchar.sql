SELECT
    t,
	ARRAY(
		SELECT
			v::varchar
		FROM
			generate_series(max(0, t - 10),  t) as v
	) AS v
	INTO test_varchar
FROM
	generate_series(1, 200) as t;


SET anyarray.similarity_type=cosine;
SELECT  t, similarity(v, '{10,9,8,7,6,5,4,3,2,1}') AS s FROM test_varchar 
	WHERE v % '{10,9,8,7,6,5,4,3,2,1}' ORDER BY s DESC, t;
SELECT  t, similarity(v, '{50,49,8,7,6,5,4,3,2,1}') AS s FROM test_varchar 
	WHERE v % '{50,49,8,7,6,5,4,3,2,1}' ORDER BY s DESC, t;

SET anyarray.similarity_type=jaccard;
SELECT  t, similarity(v, '{10,9,8,7,6,5,4,3,2,1}') AS s FROM test_varchar 
	WHERE v % '{10,9,8,7,6,5,4,3,2,1}' ORDER BY s DESC, t;
SELECT  t, similarity(v, '{50,49,8,7,6,5,4,3,2,1}') AS s FROM test_varchar 
	WHERE v % '{50,49,8,7,6,5,4,3,2,1}' ORDER BY s DESC, t;

SELECT t, v FROM test_varchar WHERE v && '{43,50}' ORDER BY t;
SELECT t, v FROM test_varchar WHERE v @> '{43,50}' ORDER BY t;
SELECT t, v FROM test_varchar WHERE v <@ '{0,1,2,3,4,5,6,7,8,9,10}' ORDER BY t;
SELECT t, v FROM test_varchar WHERE v =  '{0,1,2,3,4,5,6,7,8,9,10}' ORDER BY t;
SET anyarray.similarity_type=cosine;
SELECT t, v FROM test_varchar WHERE v %  '{0,1,2,3,4,5,6,7,8,9,10}' ORDER BY t;
SET anyarray.similarity_type=jaccard;
SELECT t, v FROM test_varchar WHERE v %  '{0,1,2,3,4,5,6,7,8,9,10}' ORDER BY t;
SET anyarray.similarity_type=overlap;
SET anyarray.similarity_threshold = 3;
SELECT t, v FROM test_varchar WHERE v %  '{0,1,2,3,4,5,6,7,8,9,10}' ORDER BY t;
RESET anyarray.similarity_threshold;

CREATE INDEX idx_test_varchar ON test_varchar USING gist (v _varchar_aa_ops);

SET enable_seqscan=off;

EXPLAIN (COSTS OFF) SELECT t, v FROM test_varchar WHERE v && '{43,50}' ORDER BY t;
EXPLAIN (COSTS OFF) SELECT t, v FROM test_varchar WHERE v @> '{43,50}' ORDER BY t;
EXPLAIN (COSTS OFF) SELECT t, v FROM test_varchar WHERE v <@ '{0,1,2,3,4,5,6,7,8,9,10}' ORDER BY t;
EXPLAIN (COSTS OFF) SELECT t, v FROM test_varchar WHERE v =  '{0,1,2,3,4,5,6,7,8,9,10}' ORDER BY t;
EXPLAIN (COSTS OFF) SELECT t, v FROM test_varchar WHERE v %  '{0,1,2,3,4,5,6,7,8,9,10}' ORDER BY t;

SELECT t, v FROM test_varchar WHERE v && '{43,50}' ORDER BY t;
SELECT t, v FROM test_varchar WHERE v @> '{43,50}' ORDER BY t;
SELECT t, v FROM test_varchar WHERE v <@ '{0,1,2,3,4,5,6,7,8,9,10}' ORDER BY t;
SELECT t, v FROM test_varchar WHERE v =  '{0,1,2,3,4,5,6,7,8,9,10}' ORDER BY t;
SET anyarray.similarity_type=cosine;
SELECT t, v FROM test_varchar WHERE v %  '{0,1,2,3,4,5,6,7,8,9,10}' ORDER BY t;
SET anyarray.similarity_type=jaccard;
SELECT t, v FROM test_varchar WHERE v %  '{0,1,2,3,4,5,6,7,8,9,10}' ORDER BY t;
SET anyarray.similarity_type=overlap;
SET anyarray.similarity_threshold = 3;
SELECT t, v FROM test_varchar WHERE v %  '{0,1,2,3,4,5,6,7,8,9,10}' ORDER BY t;
RESET anyarray.similarity_threshold;

DROP INDEX idx_test_varchar;
CREATE INDEX idx_test_varchar ON test_varchar USING gin (v _varchar_aa_ops);

SET enable_seqscan=off;

EXPLAIN (COSTS OFF) SELECT t, v FROM test_varchar WHERE v && '{43,50}' ORDER BY t;
EXPLAIN (COSTS OFF) SELECT t, v FROM test_varchar WHERE v @> '{43,50}' ORDER BY t;
EXPLAIN (COSTS OFF) SELECT t, v FROM test_varchar WHERE v <@ '{0,1,2,3,4,5,6,7,8,9,10}' ORDER BY t;
EXPLAIN (COSTS OFF) SELECT t, v FROM test_varchar WHERE v =  '{0,1,2,3,4,5,6,7,8,9,10}' ORDER BY t;
EXPLAIN (COSTS OFF) SELECT t, v FROM test_varchar WHERE v %  '{0,1,2,3,4,5,6,7,8,9,10}' ORDER BY t;

SELECT t, v FROM test_varchar WHERE v && '{43,50}' ORDER BY t;
SELECT t, v FROM test_varchar WHERE v @> '{43,50}' ORDER BY t;
SELECT t, v FROM test_varchar WHERE v <@ '{0,1,2,3,4,5,6,7,8,9,10}' ORDER BY t;
SELECT t, v FROM test_varchar WHERE v =  '{0,1,2,3,4,5,6,7,8,9,10}' ORDER BY t;
SET anyarray.similarity_type=cosine;
SELECT t, v FROM test_varchar WHERE v %  '{0,1,2,3,4,5,6,7,8,9,10}' ORDER BY t;
SET anyarray.similarity_type=jaccard;
SELECT t, v FROM test_varchar WHERE v %  '{0,1,2,3,4,5,6,7,8,9,10}' ORDER BY t;
SET anyarray.similarity_type=overlap;
SET anyarray.similarity_threshold = 3;
SELECT t, v FROM test_varchar WHERE v %  '{0,1,2,3,4,5,6,7,8,9,10}' ORDER BY t;
RESET anyarray.similarity_threshold;

