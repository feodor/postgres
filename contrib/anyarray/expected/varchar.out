SELECT
    t,
	ARRAY(
		SELECT
			v::varchar
		FROM
			generate_series(max(0, t - 10),  t) as v
	) AS v
	INTO test_varchar
FROM
	generate_series(1, 200) as t;
SET anyarray.similarity_type=cosine;
SELECT  t, similarity(v, '{10,9,8,7,6,5,4,3,2,1}') AS s FROM test_varchar 
	WHERE v % '{10,9,8,7,6,5,4,3,2,1}' ORDER BY s DESC, t;
 t  |    s     
----+----------
 10 | 0.953463
 11 | 0.953463
  9 |      0.9
 12 | 0.858116
  8 | 0.843274
  7 | 0.782624
 13 |  0.76277
  6 | 0.717137
 14 | 0.667424
  5 | 0.645497
(10 rows)

SELECT  t, similarity(v, '{50,49,8,7,6,5,4,3,2,1}') AS s FROM test_varchar 
	WHERE v % '{50,49,8,7,6,5,4,3,2,1}' ORDER BY s DESC, t;
 t  |    s     
----+----------
  8 | 0.843274
  9 |      0.8
  7 | 0.782624
 10 |  0.76277
 11 |  0.76277
  6 | 0.717137
 12 | 0.667424
  5 | 0.645497
(8 rows)

SET anyarray.similarity_type=jaccard;
SELECT  t, similarity(v, '{10,9,8,7,6,5,4,3,2,1}') AS s FROM test_varchar 
	WHERE v % '{10,9,8,7,6,5,4,3,2,1}' ORDER BY s DESC, t;
 t  |    s     
----+----------
 10 | 0.909091
 11 | 0.909091
  9 | 0.818182
 12 |     0.75
  8 | 0.727273
  7 | 0.636364
 13 | 0.615385
(7 rows)

SELECT  t, similarity(v, '{50,49,8,7,6,5,4,3,2,1}') AS s FROM test_varchar 
	WHERE v % '{50,49,8,7,6,5,4,3,2,1}' ORDER BY s DESC, t;
 t  |    s     
----+----------
  8 | 0.727273
  9 | 0.666667
  7 | 0.636364
 10 | 0.615385
 11 | 0.615385
(5 rows)

SELECT t, v FROM test_varchar WHERE v && '{43,50}' ORDER BY t;
 t  |                 v                  
----+------------------------------------
 43 | {33,34,35,36,37,38,39,40,41,42,43}
 44 | {34,35,36,37,38,39,40,41,42,43,44}
 45 | {35,36,37,38,39,40,41,42,43,44,45}
 46 | {36,37,38,39,40,41,42,43,44,45,46}
 47 | {37,38,39,40,41,42,43,44,45,46,47}
 48 | {38,39,40,41,42,43,44,45,46,47,48}
 49 | {39,40,41,42,43,44,45,46,47,48,49}
 50 | {40,41,42,43,44,45,46,47,48,49,50}
 51 | {41,42,43,44,45,46,47,48,49,50,51}
 52 | {42,43,44,45,46,47,48,49,50,51,52}
 53 | {43,44,45,46,47,48,49,50,51,52,53}
 54 | {44,45,46,47,48,49,50,51,52,53,54}
 55 | {45,46,47,48,49,50,51,52,53,54,55}
 56 | {46,47,48,49,50,51,52,53,54,55,56}
 57 | {47,48,49,50,51,52,53,54,55,56,57}
 58 | {48,49,50,51,52,53,54,55,56,57,58}
 59 | {49,50,51,52,53,54,55,56,57,58,59}
 60 | {50,51,52,53,54,55,56,57,58,59,60}
(18 rows)

SELECT t, v FROM test_varchar WHERE v @> '{43,50}' ORDER BY t;
 t  |                 v                  
----+------------------------------------
 50 | {40,41,42,43,44,45,46,47,48,49,50}
 51 | {41,42,43,44,45,46,47,48,49,50,51}
 52 | {42,43,44,45,46,47,48,49,50,51,52}
 53 | {43,44,45,46,47,48,49,50,51,52,53}
(4 rows)

SELECT t, v FROM test_varchar WHERE v <@ '{0,1,2,3,4,5,6,7,8,9,10}' ORDER BY t;
 t  |            v             
----+--------------------------
  1 | {0,1}
  2 | {0,1,2}
  3 | {0,1,2,3}
  4 | {0,1,2,3,4}
  5 | {0,1,2,3,4,5}
  6 | {0,1,2,3,4,5,6}
  7 | {0,1,2,3,4,5,6,7}
  8 | {0,1,2,3,4,5,6,7,8}
  9 | {0,1,2,3,4,5,6,7,8,9}
 10 | {0,1,2,3,4,5,6,7,8,9,10}
(10 rows)

SELECT t, v FROM test_varchar WHERE v =  '{0,1,2,3,4,5,6,7,8,9,10}' ORDER BY t;
 t  |            v             
----+--------------------------
 10 | {0,1,2,3,4,5,6,7,8,9,10}
(1 row)

SET anyarray.similarity_type=cosine;
SELECT t, v FROM test_varchar WHERE v %  '{0,1,2,3,4,5,6,7,8,9,10}' ORDER BY t;
 t  |              v               
----+------------------------------
  3 | {0,1,2,3}
  4 | {0,1,2,3,4}
  5 | {0,1,2,3,4,5}
  6 | {0,1,2,3,4,5,6}
  7 | {0,1,2,3,4,5,6,7}
  8 | {0,1,2,3,4,5,6,7,8}
  9 | {0,1,2,3,4,5,6,7,8,9}
 10 | {0,1,2,3,4,5,6,7,8,9,10}
 11 | {1,2,3,4,5,6,7,8,9,10,11}
 12 | {2,3,4,5,6,7,8,9,10,11,12}
 13 | {3,4,5,6,7,8,9,10,11,12,13}
 14 | {4,5,6,7,8,9,10,11,12,13,14}
(12 rows)

SET anyarray.similarity_type=jaccard;
SELECT t, v FROM test_varchar WHERE v %  '{0,1,2,3,4,5,6,7,8,9,10}' ORDER BY t;
 t  |             v              
----+----------------------------
  6 | {0,1,2,3,4,5,6}
  7 | {0,1,2,3,4,5,6,7}
  8 | {0,1,2,3,4,5,6,7,8}
  9 | {0,1,2,3,4,5,6,7,8,9}
 10 | {0,1,2,3,4,5,6,7,8,9,10}
 11 | {1,2,3,4,5,6,7,8,9,10,11}
 12 | {2,3,4,5,6,7,8,9,10,11,12}
(7 rows)

SET anyarray.similarity_type=overlap;
SET anyarray.similarity_threshold = 3;
SELECT t, v FROM test_varchar WHERE v %  '{0,1,2,3,4,5,6,7,8,9,10}' ORDER BY t;
 t  |                v                 
----+----------------------------------
  2 | {0,1,2}
  3 | {0,1,2,3}
  4 | {0,1,2,3,4}
  5 | {0,1,2,3,4,5}
  6 | {0,1,2,3,4,5,6}
  7 | {0,1,2,3,4,5,6,7}
  8 | {0,1,2,3,4,5,6,7,8}
  9 | {0,1,2,3,4,5,6,7,8,9}
 10 | {0,1,2,3,4,5,6,7,8,9,10}
 11 | {1,2,3,4,5,6,7,8,9,10,11}
 12 | {2,3,4,5,6,7,8,9,10,11,12}
 13 | {3,4,5,6,7,8,9,10,11,12,13}
 14 | {4,5,6,7,8,9,10,11,12,13,14}
 15 | {5,6,7,8,9,10,11,12,13,14,15}
 16 | {6,7,8,9,10,11,12,13,14,15,16}
 17 | {7,8,9,10,11,12,13,14,15,16,17}
 18 | {8,9,10,11,12,13,14,15,16,17,18}
(17 rows)

RESET anyarray.similarity_threshold;
CREATE INDEX idx_test_varchar ON test_varchar USING gist (v _varchar_aa_ops);
SET enable_seqscan=off;
EXPLAIN (COSTS OFF) SELECT t, v FROM test_varchar WHERE v && '{43,50}' ORDER BY t;
                           QUERY PLAN                            
-----------------------------------------------------------------
 Sort
   Sort Key: t
   ->  Bitmap Heap Scan on test_varchar
         Recheck Cond: (v && '{43,50}'::character varying[])
         ->  Bitmap Index Scan on idx_test_varchar
               Index Cond: (v && '{43,50}'::character varying[])
(6 rows)

EXPLAIN (COSTS OFF) SELECT t, v FROM test_varchar WHERE v @> '{43,50}' ORDER BY t;
                        QUERY PLAN                         
-----------------------------------------------------------
 Sort
   Sort Key: t
   ->  Index Scan using idx_test_varchar on test_varchar
         Index Cond: (v @> '{43,50}'::character varying[])
(4 rows)

EXPLAIN (COSTS OFF) SELECT t, v FROM test_varchar WHERE v <@ '{0,1,2,3,4,5,6,7,8,9,10}' ORDER BY t;
                                 QUERY PLAN                                 
----------------------------------------------------------------------------
 Sort
   Sort Key: t
   ->  Index Scan using idx_test_varchar on test_varchar
         Index Cond: (v <@ '{0,1,2,3,4,5,6,7,8,9,10}'::character varying[])
(4 rows)

EXPLAIN (COSTS OFF) SELECT t, v FROM test_varchar WHERE v =  '{0,1,2,3,4,5,6,7,8,9,10}' ORDER BY t;
                                QUERY PLAN                                 
---------------------------------------------------------------------------
 Sort
   Sort Key: t
   ->  Index Scan using idx_test_varchar on test_varchar
         Index Cond: (v = '{0,1,2,3,4,5,6,7,8,9,10}'::character varying[])
(4 rows)

EXPLAIN (COSTS OFF) SELECT t, v FROM test_varchar WHERE v %  '{0,1,2,3,4,5,6,7,8,9,10}' ORDER BY t;
                                QUERY PLAN                                 
---------------------------------------------------------------------------
 Sort
   Sort Key: t
   ->  Index Scan using idx_test_varchar on test_varchar
         Index Cond: (v % '{0,1,2,3,4,5,6,7,8,9,10}'::character varying[])
(4 rows)

SELECT t, v FROM test_varchar WHERE v && '{43,50}' ORDER BY t;
 t  |                 v                  
----+------------------------------------
 43 | {33,34,35,36,37,38,39,40,41,42,43}
 44 | {34,35,36,37,38,39,40,41,42,43,44}
 45 | {35,36,37,38,39,40,41,42,43,44,45}
 46 | {36,37,38,39,40,41,42,43,44,45,46}
 47 | {37,38,39,40,41,42,43,44,45,46,47}
 48 | {38,39,40,41,42,43,44,45,46,47,48}
 49 | {39,40,41,42,43,44,45,46,47,48,49}
 50 | {40,41,42,43,44,45,46,47,48,49,50}
 51 | {41,42,43,44,45,46,47,48,49,50,51}
 52 | {42,43,44,45,46,47,48,49,50,51,52}
 53 | {43,44,45,46,47,48,49,50,51,52,53}
 54 | {44,45,46,47,48,49,50,51,52,53,54}
 55 | {45,46,47,48,49,50,51,52,53,54,55}
 56 | {46,47,48,49,50,51,52,53,54,55,56}
 57 | {47,48,49,50,51,52,53,54,55,56,57}
 58 | {48,49,50,51,52,53,54,55,56,57,58}
 59 | {49,50,51,52,53,54,55,56,57,58,59}
 60 | {50,51,52,53,54,55,56,57,58,59,60}
(18 rows)

SELECT t, v FROM test_varchar WHERE v @> '{43,50}' ORDER BY t;
 t  |                 v                  
----+------------------------------------
 50 | {40,41,42,43,44,45,46,47,48,49,50}
 51 | {41,42,43,44,45,46,47,48,49,50,51}
 52 | {42,43,44,45,46,47,48,49,50,51,52}
 53 | {43,44,45,46,47,48,49,50,51,52,53}
(4 rows)

SELECT t, v FROM test_varchar WHERE v <@ '{0,1,2,3,4,5,6,7,8,9,10}' ORDER BY t;
 t  |            v             
----+--------------------------
  1 | {0,1}
  2 | {0,1,2}
  3 | {0,1,2,3}
  4 | {0,1,2,3,4}
  5 | {0,1,2,3,4,5}
  6 | {0,1,2,3,4,5,6}
  7 | {0,1,2,3,4,5,6,7}
  8 | {0,1,2,3,4,5,6,7,8}
  9 | {0,1,2,3,4,5,6,7,8,9}
 10 | {0,1,2,3,4,5,6,7,8,9,10}
(10 rows)

SELECT t, v FROM test_varchar WHERE v =  '{0,1,2,3,4,5,6,7,8,9,10}' ORDER BY t;
 t  |            v             
----+--------------------------
 10 | {0,1,2,3,4,5,6,7,8,9,10}
(1 row)

SET anyarray.similarity_type=cosine;
SELECT t, v FROM test_varchar WHERE v %  '{0,1,2,3,4,5,6,7,8,9,10}' ORDER BY t;
 t  |              v               
----+------------------------------
  3 | {0,1,2,3}
  4 | {0,1,2,3,4}
  5 | {0,1,2,3,4,5}
  6 | {0,1,2,3,4,5,6}
  7 | {0,1,2,3,4,5,6,7}
  8 | {0,1,2,3,4,5,6,7,8}
  9 | {0,1,2,3,4,5,6,7,8,9}
 10 | {0,1,2,3,4,5,6,7,8,9,10}
 11 | {1,2,3,4,5,6,7,8,9,10,11}
 12 | {2,3,4,5,6,7,8,9,10,11,12}
 13 | {3,4,5,6,7,8,9,10,11,12,13}
 14 | {4,5,6,7,8,9,10,11,12,13,14}
(12 rows)

SET anyarray.similarity_type=jaccard;
SELECT t, v FROM test_varchar WHERE v %  '{0,1,2,3,4,5,6,7,8,9,10}' ORDER BY t;
 t  |             v              
----+----------------------------
  6 | {0,1,2,3,4,5,6}
  7 | {0,1,2,3,4,5,6,7}
  8 | {0,1,2,3,4,5,6,7,8}
  9 | {0,1,2,3,4,5,6,7,8,9}
 10 | {0,1,2,3,4,5,6,7,8,9,10}
 11 | {1,2,3,4,5,6,7,8,9,10,11}
 12 | {2,3,4,5,6,7,8,9,10,11,12}
(7 rows)

SET anyarray.similarity_type=overlap;
SET anyarray.similarity_threshold = 3;
SELECT t, v FROM test_varchar WHERE v %  '{0,1,2,3,4,5,6,7,8,9,10}' ORDER BY t;
 t  |                v                 
----+----------------------------------
  2 | {0,1,2}
  3 | {0,1,2,3}
  4 | {0,1,2,3,4}
  5 | {0,1,2,3,4,5}
  6 | {0,1,2,3,4,5,6}
  7 | {0,1,2,3,4,5,6,7}
  8 | {0,1,2,3,4,5,6,7,8}
  9 | {0,1,2,3,4,5,6,7,8,9}
 10 | {0,1,2,3,4,5,6,7,8,9,10}
 11 | {1,2,3,4,5,6,7,8,9,10,11}
 12 | {2,3,4,5,6,7,8,9,10,11,12}
 13 | {3,4,5,6,7,8,9,10,11,12,13}
 14 | {4,5,6,7,8,9,10,11,12,13,14}
 15 | {5,6,7,8,9,10,11,12,13,14,15}
 16 | {6,7,8,9,10,11,12,13,14,15,16}
 17 | {7,8,9,10,11,12,13,14,15,16,17}
 18 | {8,9,10,11,12,13,14,15,16,17,18}
(17 rows)

RESET anyarray.similarity_threshold;
DROP INDEX idx_test_varchar;
CREATE INDEX idx_test_varchar ON test_varchar USING gin (v _varchar_aa_ops);
SET enable_seqscan=off;
EXPLAIN (COSTS OFF) SELECT t, v FROM test_varchar WHERE v && '{43,50}' ORDER BY t;
                           QUERY PLAN                            
-----------------------------------------------------------------
 Sort
   Sort Key: t
   ->  Bitmap Heap Scan on test_varchar
         Recheck Cond: (v && '{43,50}'::character varying[])
         ->  Bitmap Index Scan on idx_test_varchar
               Index Cond: (v && '{43,50}'::character varying[])
(6 rows)

EXPLAIN (COSTS OFF) SELECT t, v FROM test_varchar WHERE v @> '{43,50}' ORDER BY t;
                           QUERY PLAN                            
-----------------------------------------------------------------
 Sort
   Sort Key: t
   ->  Bitmap Heap Scan on test_varchar
         Recheck Cond: (v @> '{43,50}'::character varying[])
         ->  Bitmap Index Scan on idx_test_varchar
               Index Cond: (v @> '{43,50}'::character varying[])
(6 rows)

EXPLAIN (COSTS OFF) SELECT t, v FROM test_varchar WHERE v <@ '{0,1,2,3,4,5,6,7,8,9,10}' ORDER BY t;
                                    QUERY PLAN                                    
----------------------------------------------------------------------------------
 Sort
   Sort Key: t
   ->  Bitmap Heap Scan on test_varchar
         Recheck Cond: (v <@ '{0,1,2,3,4,5,6,7,8,9,10}'::character varying[])
         ->  Bitmap Index Scan on idx_test_varchar
               Index Cond: (v <@ '{0,1,2,3,4,5,6,7,8,9,10}'::character varying[])
(6 rows)

EXPLAIN (COSTS OFF) SELECT t, v FROM test_varchar WHERE v =  '{0,1,2,3,4,5,6,7,8,9,10}' ORDER BY t;
                                   QUERY PLAN                                    
---------------------------------------------------------------------------------
 Sort
   Sort Key: t
   ->  Bitmap Heap Scan on test_varchar
         Recheck Cond: (v = '{0,1,2,3,4,5,6,7,8,9,10}'::character varying[])
         ->  Bitmap Index Scan on idx_test_varchar
               Index Cond: (v = '{0,1,2,3,4,5,6,7,8,9,10}'::character varying[])
(6 rows)

EXPLAIN (COSTS OFF) SELECT t, v FROM test_varchar WHERE v %  '{0,1,2,3,4,5,6,7,8,9,10}' ORDER BY t;
                                   QUERY PLAN                                    
---------------------------------------------------------------------------------
 Sort
   Sort Key: t
   ->  Bitmap Heap Scan on test_varchar
         Recheck Cond: (v % '{0,1,2,3,4,5,6,7,8,9,10}'::character varying[])
         ->  Bitmap Index Scan on idx_test_varchar
               Index Cond: (v % '{0,1,2,3,4,5,6,7,8,9,10}'::character varying[])
(6 rows)

SELECT t, v FROM test_varchar WHERE v && '{43,50}' ORDER BY t;
 t  |                 v                  
----+------------------------------------
 43 | {33,34,35,36,37,38,39,40,41,42,43}
 44 | {34,35,36,37,38,39,40,41,42,43,44}
 45 | {35,36,37,38,39,40,41,42,43,44,45}
 46 | {36,37,38,39,40,41,42,43,44,45,46}
 47 | {37,38,39,40,41,42,43,44,45,46,47}
 48 | {38,39,40,41,42,43,44,45,46,47,48}
 49 | {39,40,41,42,43,44,45,46,47,48,49}
 50 | {40,41,42,43,44,45,46,47,48,49,50}
 51 | {41,42,43,44,45,46,47,48,49,50,51}
 52 | {42,43,44,45,46,47,48,49,50,51,52}
 53 | {43,44,45,46,47,48,49,50,51,52,53}
 54 | {44,45,46,47,48,49,50,51,52,53,54}
 55 | {45,46,47,48,49,50,51,52,53,54,55}
 56 | {46,47,48,49,50,51,52,53,54,55,56}
 57 | {47,48,49,50,51,52,53,54,55,56,57}
 58 | {48,49,50,51,52,53,54,55,56,57,58}
 59 | {49,50,51,52,53,54,55,56,57,58,59}
 60 | {50,51,52,53,54,55,56,57,58,59,60}
(18 rows)

SELECT t, v FROM test_varchar WHERE v @> '{43,50}' ORDER BY t;
 t  |                 v                  
----+------------------------------------
 50 | {40,41,42,43,44,45,46,47,48,49,50}
 51 | {41,42,43,44,45,46,47,48,49,50,51}
 52 | {42,43,44,45,46,47,48,49,50,51,52}
 53 | {43,44,45,46,47,48,49,50,51,52,53}
(4 rows)

SELECT t, v FROM test_varchar WHERE v <@ '{0,1,2,3,4,5,6,7,8,9,10}' ORDER BY t;
 t  |            v             
----+--------------------------
  1 | {0,1}
  2 | {0,1,2}
  3 | {0,1,2,3}
  4 | {0,1,2,3,4}
  5 | {0,1,2,3,4,5}
  6 | {0,1,2,3,4,5,6}
  7 | {0,1,2,3,4,5,6,7}
  8 | {0,1,2,3,4,5,6,7,8}
  9 | {0,1,2,3,4,5,6,7,8,9}
 10 | {0,1,2,3,4,5,6,7,8,9,10}
(10 rows)

SELECT t, v FROM test_varchar WHERE v =  '{0,1,2,3,4,5,6,7,8,9,10}' ORDER BY t;
 t  |            v             
----+--------------------------
 10 | {0,1,2,3,4,5,6,7,8,9,10}
(1 row)

SET anyarray.similarity_type=cosine;
SELECT t, v FROM test_varchar WHERE v %  '{0,1,2,3,4,5,6,7,8,9,10}' ORDER BY t;
 t  |              v               
----+------------------------------
  3 | {0,1,2,3}
  4 | {0,1,2,3,4}
  5 | {0,1,2,3,4,5}
  6 | {0,1,2,3,4,5,6}
  7 | {0,1,2,3,4,5,6,7}
  8 | {0,1,2,3,4,5,6,7,8}
  9 | {0,1,2,3,4,5,6,7,8,9}
 10 | {0,1,2,3,4,5,6,7,8,9,10}
 11 | {1,2,3,4,5,6,7,8,9,10,11}
 12 | {2,3,4,5,6,7,8,9,10,11,12}
 13 | {3,4,5,6,7,8,9,10,11,12,13}
 14 | {4,5,6,7,8,9,10,11,12,13,14}
(12 rows)

SET anyarray.similarity_type=jaccard;
SELECT t, v FROM test_varchar WHERE v %  '{0,1,2,3,4,5,6,7,8,9,10}' ORDER BY t;
 t  |             v              
----+----------------------------
  6 | {0,1,2,3,4,5,6}
  7 | {0,1,2,3,4,5,6,7}
  8 | {0,1,2,3,4,5,6,7,8}
  9 | {0,1,2,3,4,5,6,7,8,9}
 10 | {0,1,2,3,4,5,6,7,8,9,10}
 11 | {1,2,3,4,5,6,7,8,9,10,11}
 12 | {2,3,4,5,6,7,8,9,10,11,12}
(7 rows)

SET anyarray.similarity_type=overlap;
SET anyarray.similarity_threshold = 3;
SELECT t, v FROM test_varchar WHERE v %  '{0,1,2,3,4,5,6,7,8,9,10}' ORDER BY t;
 t  |                v                 
----+----------------------------------
  2 | {0,1,2}
  3 | {0,1,2,3}
  4 | {0,1,2,3,4}
  5 | {0,1,2,3,4,5}
  6 | {0,1,2,3,4,5,6}
  7 | {0,1,2,3,4,5,6,7}
  8 | {0,1,2,3,4,5,6,7,8}
  9 | {0,1,2,3,4,5,6,7,8,9}
 10 | {0,1,2,3,4,5,6,7,8,9,10}
 11 | {1,2,3,4,5,6,7,8,9,10,11}
 12 | {2,3,4,5,6,7,8,9,10,11,12}
 13 | {3,4,5,6,7,8,9,10,11,12,13}
 14 | {4,5,6,7,8,9,10,11,12,13,14}
 15 | {5,6,7,8,9,10,11,12,13,14,15}
 16 | {6,7,8,9,10,11,12,13,14,15,16}
 17 | {7,8,9,10,11,12,13,14,15,16,17}
 18 | {8,9,10,11,12,13,14,15,16,17,18}
(17 rows)

RESET anyarray.similarity_threshold;
